### PR TITLE
add option to limit max number of game instances to be started

### PR DESCRIPTION
**Pull request by Fusselwurm**
currently, it's far too easy to effectively kill a host by starting too many instances ^^

### CODE REVIEW

File: lib/manager.js [hunk 1]
```diff
@@ -82,14 +96,35 @@ Manager.prototype.load = function () {
       console.error('Manager load error: ' + e)
     }
 
-    self.getServers().map(function (server) {
-      if (server.auto_start) {
-        server.start()
-      }
-    })
+    self.getServers()
+      .filter(server => server.auto_start)
```
Comment (Dahlgren) [inline lib/manager.js]: This will make old Node.js versions go 💥

Comment (Fusselwurm) [reply lib/manager.js]: do we really need to be compatible with NodeJS<4 :sob:

File: lib/manager.js [hunk 2]
```diff
@@ -3,6 +3,11 @@ var fs = require('fs')
 
 var Server = require('./server')
 
+var maxInstances = require('../config').maxInstances
+if (isNaN(maxInstances)) {
+  throw new Error('missing config option "maxInstances"')
```
[the remark this answers is not in the document]
Comment (Fusselwurm) [reply lib/manager.js]: right! i'll just take Infinity as fallback if maxInstances is falsy. Hoping no-one will set it to zero.

File: lib/manager.js [hunk 3]
```diff
@@ -3,6 +3,7 @@ var fs = require('fs')
 
 var Server = require('./server')
 
+var maxInstances = require('../config').maxInstances || Infinity
```
Comment (Dahlgren) [inline lib/manager.js]: config is injected into creation of manager

**Pull request closed without merging**